src/test/java/seedu/address/logic/commands/EditModuleClassCommandTest.java

```diff
@@ -0,0 +1,170 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.DESC_CS2100_TUTORIAL;
+import static seedu.address.logic.commands.CommandTestUtil.DESC_CS2103T_TUTORIAL;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_NAME_CS2030_TUTORIAL;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showModuleClassAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_MODULE_CLASS;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_MODULE_CLASS;
+import static seedu.address.testutil.TypicalTutorsPet.getTypicalTutorsPet;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.EditModuleClassCommand.EditModuleClassDescriptor;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.TutorsPet;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.moduleclass.ModuleClass;
+import seedu.address.testutil.EditModuleClassDescriptorBuilder;
+import seedu.address.testutil.ModuleClassBuilder;
+import seedu.address.testutil.TypicalModuleClass;
+
+/**
+ * Contains integration tests (interaction with the Model, UndoCommand and RedoCommand) and unit tests for
```

Comment:
Seeing this template always makes me feel like they have already foreshadowed that we should do undo/redo features.
```suggestion
 * Contains integration tests (interaction with the Model) and unit tests for
```